lib/sorbet-rails/typed_params.rb

```diff
@@ -0,0 +1,17 @@
+# typed: true
+require 'sorbet-coerce'
+
+module TypedParams
+  # A sorbet-coerce wrapper for coercing typed controller parameters
+  define_singleton_method(:[]) do |type|
+    Class.new do
+      define_method(:extract!) do |params|
+        begin
+          T::Coerce[type].new.from(params.permit!.to_h, raise_coercion_error: true)
```

Comment:
Do we want to make `raise_coercion_error` an option on TypedParams as well?

Reply:
Hmm yeah, we might as well!